### PR TITLE
handle connection errors @ `MssqlDriver`.

### LINKED ISSUE #836: How to prevent Kysely from crashing node app if database goes down
Hi All,

I am trying to incorporate Kysely into a larger Node.js application to write data to a MSSQL database. I create a instance of Kysely exactly like in the doc as shown below and execute inserts with an error catch handler however when the SQL database is not reachable the whole node application crashes. 

How can I best handle situation where the SQL database is unreachable without it crashing the whole app?

```Insert
db
        .insertInto('commandQueue').values({
            command,
            status: 'pending',
            terminal_id: terminalId,
            server_flag: msgId,
        })
        .execute()
        .then(() => console.log(date.toLocaleTimeString(),`[${terminalId}]`, `command ${command} queued`))
        .catch(e => console.log(e))
```

Error
```
TimeoutError: Failed to connect to localhost:1433 - Could not connect (sequence)
    at /Users/anthonyphan/repos/nex-gps/node_modules/tarn/dist/PendingOperation.js:14:27
    at runNextTicks (node:internal/process/task_queues:60:5)
    at listOnTimeout (node:internal/timers:538:9)
    at processTimers (node:internal/timers:512:7)
    at async MssqlDriver.acquireConnection (/Users/anthonyphan/repos/nex-gps/node_modules/kysely/dist/cjs/dialect/mssql/mssql-driver.js:40:16)
    at async RuntimeDriver.acquireConnection (/Users/anthonyphan/repos/nex-gps/node_modules/kysely/dist/cjs/driver/runtime-driver.js:46:28)
    at async DefaultConnectionProvider.provideConnection (/Users/anthonyphan/repos/nex-gps/node_modules/kysely/dist/cjs/driver/default-connection-provider.js:10:28)
    at async DefaultQueryExecutor.executeQuery (/Users/anthonyphan/repos/nex-gps/node_modules/kysely/dist/cjs/query-executor/query-executor-base.js:36:16)
    at async InsertQueryBuilder.execute (/Users/anthonyphan/repos/nex-gps/node_modules/kysely/dist/cjs/query-builder/insert-query-builder.js:528:24)
/Users/anthonyphan/repos/nex-gps/node_modules/kysely/dist/cjs/dialect/mssql/mssql-driver.js:40
        return await this.#pool.acquire().promise;
               ^
TimeoutError: Failed to connect to localhost:1433 - Could not connect (sequence)
    at /Users/anthonyphan/repos/nex-gps/node_modules/tarn/dist/PendingOperation.js:14:27
    at async MssqlDriver.acquireConnection (/Users/anthonyphan/repos/nex-gps/node_modules/kysely/dist/cjs/dialect/mssql/mssql-driver.js:40:16)
    at async RuntimeDriver.acquireConnection (/Users/anthonyphan/repos/nex-gps/node_modules/kysely/dist/cjs/driver/runtime-driver.js:46:28)
    at async DefaultConnectionProvider.provideConnection (/Users/anthonyphan/repos/nex-gps/node_modules/kysely/dist/cjs/driver/default-connection-provider.js:10:28)
    at async DefaultQueryExecutor.executeQuery (/Users/anthonyphan/repos/nex-gps/node_modules/kysely/dist/cjs/query-executor/query-executor-base.js:36:16)
    at async InsertQueryBuilder.execute (/Users/anthonyphan/repos/nex-gps/node_modules/kysely/dist/cjs/query-builder/insert-query-builder.js:528:24)
[nodemon] app crashed - waiting for file changes before starting...
```

Creating Kysely instance

```
import { Database } from './types.ts' // this is the Database interface we defined earlier
import * as tedious from 'tedious'
import * as tarn from 'tarn'
import { Kysely, MssqlDialect } from 'kysely'

const dialect = new MssqlDialect({
  tarn: {
    ...tarn,
    options: {
      min: 0,
      max: 10,
    },
  },
  tedious: {
    ...tedious,
    connectionFactory: () => new tedious.Connection({
      authentication: {
        options: {
          password: 'password',
          userName: 'username',
        },
        type: 'default',
      },
      options: {
        database: 'some_db',
        port: 1433,
        trustServerCertificate: true,
      },
      server: 'localhost',
    }),
  },
})

// Database interface is passed to Kysely's constructor, and from now on, Kysely 
// knows your database structure.
// Dialect is passed to Kysely's constructor, and from now on, Kysely knows how 
// to communicate with your database.
export const db = new Kysely<Database>({
  dialect,
})
```

### PR DESCRIPTION
**Pull request by igalklebanov**
fixes #836

### CODE REVIEW

File: src/dialect/mssql/mssql-driver.ts [hunk 1]
```diff
@@ -104,6 +100,11 @@ class MssqlConnection implements DatabaseConnection {
   constructor(connection: TediousConnection, tedious: Tedious) {
     this.#connection = connection
     this.#tedious = tedious
+
+    this.#connection.on('error', console.error)
```
Comment (igalklebanov) [inline src/dialect/mssql/mssql-driver.ts:104]: are we cool with logging these?

File: src/dialect/mssql/mssql-driver.ts [hunk 2]
```diff
@@ -132,6 +133,21 @@ class MssqlConnection implements DatabaseConnection {
     )
   }
 
+  async connect(): Promise<this> {
+    await new Promise((resolve, reject) => {
+      this.#connection.connect((error) => {
+        if (error) {
+          console.error(error)
```
Comment (igalklebanov) [inline src/dialect/mssql/mssql-driver.ts:140]: are we cool with logging these? normally underlying drivers do..

**Review by koskimas — APPROVED** [review]
LGTM

File: src/dialect/mssql/mssql-driver.ts (hunk 1 above)
Comment (koskimas) [reply src/dialect/mssql/mssql-driver.ts:104]: That's fine.

File: src/dialect/mssql/mssql-driver.ts (hunk 2 above)
Comment (koskimas) [reply src/dialect/mssql/mssql-driver.ts:140]: Yep

**Pull request merged**